llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -4995,12 +4995,16 @@ void llvm::UpgradeIntrinsicCall(CallBase *CI, Function *NewFn) {
     break;
 
   case Intrinsic::ctlz:
-  case Intrinsic::cttz:
-    assert(CI->arg_size() == 1 &&
+  case Intrinsic::cttz: {
+    assert((CI->arg_size() <= 2) &&
```

Comment:
I think the right way to handle this is `if (CI->arg_size() != 1) { DefaultCase(); return; }` as seen in some of the other cases. We only want to take this code path for upgrading the one argument case, otherwise we should perform a remangling upgrade.